classes/plugins/importexport/PKPImportExportDeployment.inc.php

```diff
@@ -252,6 +256,11 @@ function removeImportedObjects($assocType) {
 	 * @param $user User
 	 */
 	function setUser($user) {
+		$registeredUser = Registry::get('user', true, null);
+		if (!isset($registeredUser)) {
+			Registry::set('user', $user);
```

Comment:
What is this used for?

Reply:
This is a way to try to reconcile with `$request->getUser()` calls from within classes used by import processes within a CLI import process.

Reply:
@asmecher I removed it, and I did not had any issues importing or exporting. I am sure I used it for the reason I described above, but I will remove it for now and see how that goes.

Reply:
@asmecher for example the https://github.com/pkp/pkp-lib/blob/master/classes/services/PKPSubmissionFileService.inc.php#L321 when importing using the CLI tool (`php importexport.php NativeImportExportPlugin import --xmlFileName ../test_issues_export_1.xml --journal_path ts --user_name admin`) 

Perhaps we could pass an optional `user` param to the function (or the ServiceClass constructor) and use the `$request->getUser()` only if we have not passed any specific user. @NateWr what do you think?

Reply:
The optional parameter solution appeals to me more than the base class -- but there's also an argument to be made for an abstraction between the Request and the Service, perhaps like the Deployment class, so that both web-based or command-line-based invocations can provide this kind of context in the same fashion. But let's see what @NateWr thinks.

Reply:
(@NateWr, do you have an opinion on this?)